packages/typespec-client-generator-core/design-docs/flags.md

```diff
@@ -0,0 +1,35 @@
+# TCGC `tspconfig.yaml` flags
+
+## 1. `generate-protocol-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@protocolAPI` is not set for an operation. Default value is `true`.
+
+## 2. `generate-convenience-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@convenientAPI` is not set for an operation. Default value is `true`.
+
+## 3. `package-name`
```

Comment:
I wasn't aware we had a `package-name` in TCGC, when I asks for https://github.com/microsoft/typespec/issues/5758

Yeah, we can use this, if we agree there would be no need to setting a version.

Though unsure whether there would be another `package-description` though.

Reply:
Python is fine to add `package-version/package-description` or not since they are not necessary configurations for a python package.
But I think maybe we could add these configs in case users want to set them when calling emitter.

nit: 
Python could consume `package-verison` at [here](https://github.com/microsoft/typespec/blob/ba241e8e5561e3b99d4a7b6195aa193c35cbae63/packages/http-client-python/generator/pygen/codegen/templates/version.py.jinja2#L4) and `package-description` at [here](https://github.com/microsoft/typespec/blob/ba241e8e5561e3b99d4a7b6195aa193c35cbae63/packages/http-client-python/generator/pygen/codegen/templates/packaging_templates/setup.py.jinja2#L27).

Reply:
For now package name, version and description would be useful for JS. And the package name is required but others are optional in Azure scope.

Reply:
the `package-name` will be soon deprecated in this pr: https://github.com/Azure/typespec-azure/pull/2161